### PR TITLE
Add molecules aa1 to implement 1

### PR DESCRIPTION
**Pull request by pinterid**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Documentation (non-breaking change which adds documentation)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently there is no footer component. #21 
- Currently there is no navbar component. #22 
- Currently there is no login form component. #23 
- Currently there is no register form component. #24 
- Currently there is no upload modal component. #25
- Currently there is no user action card component. #26 

**What is the new behavior (if this is a feature change)?**
- Now there is a footer component. #21  
- Now there is a navbar component. #22 
- Now there is a login form component. #23 
- Now there is a register form component. #24 
- Now there is a upload modal component. #25
- Now there is a user action card component. #26 

**Other information**:
- Ref: 🐍

### CODE REVIEW

File: src/components/molecules/Footer/index.jsx [hunk 1]
```diff
@@ -0,0 +1,265 @@
+//#region > Imports
+//> React
+// Contains all the functionality necessary to define React components
+import React from "react";
+// DOM bindings for React Router
+import { Link, withRouter } from "react-router-dom";
+
```
Comment (schettn) [inline src/components/molecules/Footer/index.jsx]:
```suggestion
```

File: src/components/molecules/Footer/index.jsx [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+  MDBRow,
+  MDBCol,
+  MDBContainer,
+  MDBIcon,
+  MDBBtn,
+} from "mdbreact";
+
+//> CSS
+import "./footer.scss";
+
```
Comment (schettn) [inline src/components/molecules/Footer/index.jsx]:
```suggestion
```

File: src/components/molecules/Navbar/index.jsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
+//#region > Imports
+//> React
+// Contains all the functionality necessary to define React components
+import React from "react";
+// DOM bindings for React Router
+import { Link, withRouter } from "react-router-dom";
+// React PropTypes
+import PropTypes from "prop-types";
+
```
Comment (schettn) [inline src/components/molecules/Navbar/index.jsx]:
```suggestion
```

File: src/components/molecules/Navbar/index.jsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  MDBDropdownItem,
+  MDBDropdownToggle,
+  MDBDropdownMenu,
+  MDBSmoothScroll,
+  MDBBtn,
+} from "mdbreact";
+
+//> Images
+import SNEKLogo from "../../../assets/navigation/logo.png";
+
```
Comment (schettn) [inline src/components/molecules/Navbar/index.jsx]:
```suggestion
```

File: src/components/molecules/UserActionCard/index.jsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
+//#region > Imports
+//> React
+// Contains all the functionality necessary to define React components
+import React from "react";
+// React PropTypes
+import PropTypes from "prop-types";
+
```
Comment (schettn) [inline src/components/molecules/UserActionCard/index.jsx]:
```suggestion
```

File: src/components/molecules/UserActionCard/index.jsx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+//> Components
+import { RegisterForm, LoginForm } from "../forms";
+
+//> Images
+import { ReactComponent as SvgSoftware } from "../../../assets/header/dev.svg";
+import { ReactComponent as SvgMedia } from "../../../assets/header/media.svg";
+
+//> CSS
+import "./useractioncard.scss";
```
Comment (schettn) [inline src/components/molecules/UserActionCard/index.jsx]:
```suggestion
//> Components
import { RegisterForm, LoginForm } from "../forms";
//> Images
import { ReactComponent as SvgSoftware } from "../../../assets/header/dev.svg";
import { ReactComponent as SvgMedia } from "../../../assets/header/media.svg";
//> CSS
import "./useractioncard.scss";
```

File: src/components/molecules/modals/UploadModal/index.jsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
+//#region > Imports
+//> React
+// Contains all the functionality necessary to define React components
+import React from "react";
+// Contains the functionality for uploading a file
+import Dropzone from "react-dropzone";
+
```
Comment (schettn) [inline src/components/molecules/modals/UploadModal/index.jsx]:
```suggestion
```

File: src/components/molecules/modals/UploadModal/index.jsx [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+    loading: false,
+    error: [],
+  };
+
+  onDrop = async (files) => {
+    if (files.length > 0) {
+      this.setState({
+        error: [],
+        loading: true,
+      });
```
Comment (schettn) [inline src/components/molecules/modals/UploadModal/index.jsx:28]:
```suggestion
      });
      
```

File: src/components/molecules/modals/UploadModal/index.jsx [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+      this.props.uploadTalk(files[0]).then(() => {
+        this.setState({
+          loading: false,
+        });
```
Comment (schettn) [inline src/components/molecules/modals/UploadModal/index.jsx:33]:
```suggestion
        });
        
```

**Review by schettn — CHANGES_REQUESTED** [review]
I left some suggestions.

**Review by schettn — APPROVED** [review]
🚀

File: src/components/molecules/Footer/index.jsx [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+import "./footer.scss";
+//> Images
+// SNEK Logo
+import Logo from "../../../assets/navigation/logo.png";
+//#endregion
+
+//> Local data
+// Slogans
+/** @todo Change to uppercase */
+//#region > Constant Variables
```
Comment (pinterid) [inline src/components/molecules/Footer/index.jsx]:
```suggestion
//#region > Constant Variables
//> Local data
// Slogans
/** @todo Change to uppercase */
```

File: src/components/molecules/forms/LoginForm/index.jsx [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+    event.stopPropagation();
+
+    let errors = [];
+
+    if (this.state.login_username === "") {
+      errors.push({
+        code: 20,
+        weight: 10,
+      });
+    }
```
Comment (pinterid) [inline src/components/molecules/forms/LoginForm/index.jsx:104]:
```suggestion
    }
    
```

File: src/components/molecules/forms/LoginForm/index.jsx [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+      this.setState({
+        errors,
+      });
+    } else {
+      // Proceed to login
+      const result = await this.props.globalFunctions.login(
+        this.state.login_username,
+        this.state.login_password
+      );
+      console.log(result);
```
Comment (pinterid) [inline src/components/molecules/forms/LoginForm/index.jsx]:
```suggestion

      console.log(result);
      
```

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 13]
```diff
@@ -0,0 +1,805 @@
...
+    }
+  };
+
+  removeSource = (id) => {
+    let sourceList = this.state.sourceList.filter(function (obj) {
+      return obj.id !== id;
+    });
+    let usernames = this.state.usernames.filter(function (obj) {
+      return obj.id !== id;
+    });
```
Comment (pinterid) [inline src/components/molecules/forms/RegisterForm/index.jsx:195]:
```suggestion
    });
    
```

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 14]
```diff
@@ -0,0 +1,805 @@
...
+              return false;
+            }
+          });
+          if (innerRtn.includes(true)) {
+            return true;
+          } else {
+            return false;
+          }
+        }
+      });
```
Comment (pinterid) [inline src/components/molecules/forms/RegisterForm/index.jsx:238]:
```suggestion
      });
      
```

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 15]
```diff
@@ -0,0 +1,805 @@
...
+
+      this.setState({
+        errors,
+      });
+    }
+  };
+
+  // Handle sumbit of register form
+  handleSubmit = async () => {
+    console.log("Handle submit");
```
Comment (pinterid) [inline src/components/molecules/forms/RegisterForm/index.jsx]:
```suggestion
    console.log("Handle submit");
    
```

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 16]
```diff
@@ -0,0 +1,805 @@
...
+    let errors = [];
+
+    // Check if passwords match
+    if (password1 !== "" && password2 !== "" && password1 !== password2) {
+      errors.push({
+        code: 1,
+        msg: "Your passwords do not match.",
+        weight: 10,
+      });
+    }
```
Comment (pinterid) [inline src/components/molecules/forms/RegisterForm/index.jsx:319]:
```suggestion
    }
    
```

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 17]
```diff
@@ -0,0 +1,805 @@
...
+    if (sourceList.length === 0) {
+      errors.push({
+        code: 2,
+        msg: "No platforms are connected.",
+        weight: 10,
+      });
+    }
```
Comment (pinterid) [inline src/components/molecules/forms/RegisterForm/index.jsx:327]:
```suggestion
    }
    
```

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 18]
```diff
@@ -0,0 +1,805 @@
...
+    if (firstname === "") {
+      errors.push({
+        code: 3,
+        msg: "Please enter your first name.",
+        weight: 8,
+      });
+    }
```
Comment (pinterid) [inline src/components/molecules/forms/RegisterForm/index.jsx:335]:
```suggestion
    }
    
```

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 19]
```diff
@@ -0,0 +1,805 @@
...
+    if (lastname === "") {
+      errors.push({
+        code: 4,
+        msg: "Please enter your last name.",
+        weight: 8,
+      });
+    }
```
Comment (pinterid) [inline src/components/molecules/forms/RegisterForm/index.jsx:343]:
```suggestion
    }
    
```

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 20]
```diff
@@ -0,0 +1,805 @@
...
+    if (email === "") {
+      errors.push({
+        code: 5,
+        msg: "Please enter your email.",
+        weight: 9,
+      });
+    }
```
Comment (pinterid) [inline src/components/molecules/forms/RegisterForm/index.jsx:351]:
```suggestion
    }
    
```

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 21]
```diff
@@ -0,0 +1,805 @@
...
+    if (username === "") {
+      errors.push({
+        code: 6,
+        msg: "Please select a username from the list above.",
+        weight: 10,
+      });
+    }
```
Comment (pinterid) [inline src/components/molecules/forms/RegisterForm/index.jsx:359]:
```suggestion
    }
    
```

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 22]
```diff
@@ -0,0 +1,805 @@
...
+    if (password1 === "") {
+      errors.push({
+        code: 7,
+        msg: "Please enter a password.",
+        weight: 10,
+      });
+    }
```
Comment (pinterid) [inline src/components/molecules/forms/RegisterForm/index.jsx:367]:
```suggestion
    }
    
```

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 23]
```diff
@@ -0,0 +1,805 @@
...
+    if (password2 === "") {
+      errors.push({
+        code: 8,
+        msg: "Please repeat your password.",
+        weight: 10,
+      });
+    }
```
Comment (pinterid) [inline src/components/molecules/forms/RegisterForm/index.jsx:375]:
```suggestion
    }
    
```

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 24]
```diff
@@ -0,0 +1,805 @@
...
+
+    if (errors.length === 0) {
+      this.setState(
+        {
+          loading: true,
+        },
+        () => {
+          // Cache data
+          let cache = {};
+          console.log("Register", this.state);
```
Comment (pinterid) [inline src/components/molecules/forms/RegisterForm/index.jsx]:
```suggestion

          console.log("Register", this.state);
          
```

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 25]
```diff
@@ -0,0 +1,805 @@
...
+          let registrationData = {
+            sources: sourceList,
+            username,
+            email,
+            first_name: firstname,
+            last_name: lastname,
+            gift_code: promoCode && code !== "" ? code : null,
+            password: password1,
+          };
```
Comment (pinterid) [inline src/components/molecules/forms/RegisterForm/index.jsx:402]:
```suggestion
          };
          
```

**Review by pinterid — COMMENTED** [review]
Some changes have to be done.

File: src/components/molecules/forms/LoginForm/index.jsx [hunk 26]
```diff
@@ -0,0 +1,215 @@
...
+        errors,
+      });
+    } else {
+      // Proceed to login
+      const result = await this.props.globalFunctions.login(
+        this.state.login_username,
+        this.state.login_password
+      );
+
+      console.log(result);
```
Comment (kleberbaum) [inline src/components/molecules/forms/LoginForm/index.jsx]: Please add a anker.

File: src/components/molecules/UserActionCard/useractioncard.scss [hunk 27]
```diff
@@ -0,0 +1,208 @@
...
+      li {
+        margin-bottom: 0 !important;
+      }
+
+      li:not(.disabled) {
+        span {
+          color: #495057;
+        }
+      }
+
```
Comment (kleberbaum) [inline src/components/molecules/UserActionCard/useractioncard.scss]: Please remove newline.

File: src/components/molecules/forms/LoginForm/index.jsx [hunk 28]
```diff
@@ -0,0 +1,215 @@
...
+      if (result) {
+        this.setState(
+          {
+            loginFail: false,
+          },
+          () => this.props.handleLogin(result)
+        );
+      } else {
+        // Login fail
+        //handleLogin(false);
```
Comment (kleberbaum) [inline src/components/molecules/forms/LoginForm/index.jsx]: Please remove this line.
```suggestion
```

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 29]
```diff
@@ -0,0 +1,816 @@
...
+      });
+    }
+
+    if (errors.length === 0) {
+      this.setState(
+        {
+          loading: true,
+        },
+        () => {
+          console.log("Register", this.state);
```
Comment (kleberbaum) [inline src/components/molecules/forms/RegisterForm/index.jsx]: Please add a anker.

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 30]
```diff
@@ -0,0 +1,816 @@
...
+        gitlab_servers,
+      });
+    }
+  };
+
+  addGitLab = () => {
+    const username = this.state.gitlab_username;
+    const server = this.state.gitlab_server;
+
+    console.log(username, server);
```
Comment (kleberbaum) [inline src/components/molecules/forms/RegisterForm/index.jsx]: Please add a anker.

File: src/components/molecules/forms/RegisterForm/index.jsx [hunk 31]
```diff
@@ -0,0 +1,816 @@
...
+
+      this.setState({
+        errors,
+      });
+    }
+  };
+
+  // Handle sumbit of register form
+  handleSubmit = async () => {
+    console.log("Handle submit");
```
Comment (kleberbaum) [inline src/components/molecules/forms/RegisterForm/index.jsx]: Please add a anker.

**Review by kleberbaum — CHANGES_REQUESTED** [review]
Should be ok to merge after a few changes.

**Review by kleberbaum — APPROVED** [review]
Should be ok to merge.

**Pull request merged**